common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/exceptions/IngestorException.scala

```diff
@@ -0,0 +1,8 @@
+package uk.ac.wellcome.pipeline_storage.exceptions
+
+case class IngestorException(e: Throwable) extends Exception(e.getMessage)
+
+case object IngestorException {
```

Comment:
What's the benefit of this pattern?

Reply:
What part do you mean? If it is that it is a `case object` rather than standard `object` I agree that looks unnecessary, or do you mean something else?

Reply:
Ah, no, just wrapping the exceptions in a case class that doesn't seem to do much else.

Reply:
I guess the difference is that we can specify a specific exception type (rather than generic Exception), and can also pattern match on it, but not entirely sure on the original intention.

On a more general note, error handling in our code is very variable, and I think at some point might make sense to come up with some common patterns that we use everywhere.

Reply:
Yip - definitely think that it would be good to do that as context switching / inventing another way takes up unnecessary time on what is a pretty well trodden domain.